app/src/main/java/com/kamwithk/ankiconnectandroid/request_parsers/Parser.java

```diff
@@ -51,6 +51,77 @@ public static String getNoteQuery(JsonObject raw_data) {
         return raw_data.get("params").getAsJsonObject().get("query").getAsString();
     }
 
+    public static long getNoteId(JsonObject raw_data) {
+        return raw_data.get("params").getAsJsonObject().get("note").getAsJsonObject().get("id").getAsLong();
+    }
+
+    public static Map<String, String> getUpdateNoteFieldsFields(JsonObject raw_data) {
+        Type fieldType = new TypeToken<Map<String, String>>() {}.getType();
+        return gson.fromJson(raw_data.get("params").getAsJsonObject().get("note").getAsJsonObject().get("fields"), fieldType);
+    }
+
+    public static Map<String, byte[]> getUpdateNoteFilesToAdd(JsonObject raw_data) {
```

Comment:
What do you think about:
In Parser.java, just have a very dumb method `ArrayList<JsonObject> getUpdateNoteFieldsMedia(JsonObject raw_data)` where the JsonObjects are JsonArrays, and then all the work that's currently done in `getUpdateNoteFilesToAdd()` and `getUpdateNoteMediaFieldsToFilenames()` can be moved into `IntegratedAPI.updateNoteFields()`?

A more sustainable solution IMO would be to make a small class that encodes the audio/video/picture parameter from [here](https://github.com/FooSoft/anki-connect#addnote) and then have the parser parse into that, but tbh I don't have much Java experience and it'd be a lot of work if I'm the only one who's going to be using it, maybe consider it if someone wants to add media support to addNote though.

Reply:
I think this is fine as long as you mention in a comment that there could be work for improvement. If the storing media files can be moved to the function as mentioned earlier (lines 90-102), then getting the parser to parse the individual components shouldn't be difficult to do anyways.

Reply:
Ended up creating a RequestMedia class for the parser to parse each audio/video/picture into, since it was easier than sending around the metadata to be honest. Let me know what you think.